pkg/clone/checksum.go

```diff
@@ -91,6 +98,39 @@ func (cmd *Checksum) reportDiffs(diffs []Diff) {
 	}
 }
 
+func (cmd *Checksum) readLag(ctx context.Context, target *sql.DB) (time.Duration, error) {
+	retry := RetryOptions{
+		Limiter:       nil, // will we ever use concurrency limiter again? probably not?
+		AcquireMetric: readLimiterDelay.WithLabelValues("target"),
+		MaxRetries:    cmd.ReadRetries,
+		Timeout:       cmd.ReadTimeout,
+	}
+
+	lag := time.Hour
+	err := Retry(ctx, retry, func(ctx context.Context) error {
+		stmt := fmt.Sprintf("SELECT time FROM %s WHERE task = ?", cmd.HeartbeatTable)
+		row := target.QueryRowContext(ctx, stmt, cmd.TaskName)
+		var lastHeartbeat time.Time
+		err := row.Scan(&lastHeartbeat)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				// We haven't received the first heartbeat yet, maybe we're an hour behind, who knows?
```

Comment:
Does it make sense to assume a large (1 hr) replication lag? It seems to me that if it is more than a few 10s of seconds then something is terribly wrong?

Reply:
No it can be really really large. With ODS databases we sometimes have several days worth of replication lag. This is usually directly after an incredibly heavy write session, like the first best effort clone. It takes a lot less time to recover several days worth of lag fortunately, only a few hours.